examples/erc1155/lib.rs

```diff
@@ -40,6 +40,38 @@ pub type TokenId = u128;
 
 type Balance = <ink_env::DefaultEnvironment as ink_env::Environment>::Balance;
 
+// The ERC-1155 error types.
+#[derive(Debug, PartialEq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum Error {
+    /// This token ID has not yet been created by the contract.
+    UnexistentToken,
+    /// The caller tried to sending tokens to the zero-address (0x00).
+    ZeroAddressTransfer,
+    /// The caller is not approved to transfer tokens on behalf of the account.
+    NotApproved,
+    /// The account does not have enough funds to complete the transfer.
+    InsufficientBalance,
+    /// An account does not need to approve themselves to transfer tokens.
+    SelfApproval,
+    /// The number of tokens being transferred does not match the specified number of transfers.
+    BatchTransferMismatch,
+}
+
+// The ERC-1155 result types.
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Evaluate `$x:expr` and if not true return `Err($y:expr)`.
+///
+/// Used as `ensure!(expression_to_ensure, expression_to_return_on_false)`.
+macro_rules! ensure {
+    ( $x:expr, $y:expr $(,)? ) => {{
```

Comment:
please strife for better names also in macros.

Reply:
Are you referring to the macro name, or the variable names?

If it's the macro name, I think it's fine. It comes from Substrate's [`frame_support`](https://crates.parity.io/frame_support/macro.ensure.html) and will be immediately familiar to those coming from runtime development.

As for the variables, they're really short lived given the size of the macro, so I think they're also fine as is.

Reply:
I am referring to `x` and `y`. We should really care for stating of intent with names even in small scopes especially since this piece of code serves as an example. ;)